engine/lib/base_jobmaster.go

```diff
@@ -79,6 +81,7 @@ type JobMasterImpl interface {
 
 	Workload() model.RescUnit
 	OnJobManagerMessage(topic p2p.Topic, message interface{}) error
+	OnOpenAPIInitialized(apiGroup *gin.RouterGroup)
```

Comment:
Why `JobMasterImpl`  has the `OnOpenAPIInitialized ` as the `BaseJobMaster `?

Reply:
`BaseJobMaster.OnOpenAPIInitialized` is triggered from the perspective of executor, and `JobMasterImpl.OnOpenAPIInitialized` does the real work that passes router group, they are two functions actually. Maybe we can use two names to make it easier to understand.

Reply:
`BaseJobMaster.OnOpenAPIInitialized` is triggered from the perspective of executor, and `JobMasterImpl.OnOpenAPIInitialized` does the real work that passes router group, they are two functions actually. Maybe we can use two names to make it easier to understand.

Reply:
Can add a comment to this function that user logic must not access `apiGroup` after this function returns, otherwise data race could happen.